### PR TITLE
Added error if multimodule passed to single-module gRPC method.

### PR DESCRIPTION
**Pull request by RIg410**

### CODE REVIEW

File: services/src/compiler.rs [hunk 1]
```diff
@@ -47,10 +48,21 @@ where
     ) -> Result<Result<Vec<u8>, String>, Status> {
         let source_file_data = request.into_inner();
         let address = convert_address(&source_file_data.address)?;
+        let mut source = HashMap::with_capacity(1);
+        source.insert("source".to_owned(), source_file_data.text);
         Ok(self
             .compiler
-            .compile(&source_file_data.text, Some(address))
-            .map_err(|err| err.to_string()))
+            .compile_source_map(source, Some(address))
```
Comment (mkurnikov) [inline services/src/compiler.rs:55]: `compile_source_map` -> `compile_with_source_map` ?

File: services/src/compiler.rs [hunk 2]
```diff
@@ -47,10 +48,21 @@ where
...
+            .map_err(|err| err.to_string())
+            .and_then(|bytecode_map| {
+                if bytecode_map.len() > 1 {
+                    Err("Unsupported multiple modules file.".to_owned())
```
Comment (mkurnikov) [inline services/src/compiler.rs:59]: `to_owned` -> `to_string` ?

File: services/tests/compiler.rs [hunk 3]
```diff
@@ -201,3 +195,13 @@ async fn test_compilation_error_on_expected_an_expression_term() {
...
 }
+
+#[tokio::test]
+async fn test_compilation_with_multiple_models() {
+    let source_text = r#"
+            module A {}
+            module B {}
+        "#;
+    let compilation_result = compile_source_file(source_text).await.unwrap().into_inner();
+    assert!(compilation_result.errors[0].contains("Unsupported multiple modules file."));
```
Comment (mkurnikov) [inline services/tests/compiler.rs:206]: Check whether `errors.len() == 0` ?

Comment (fzzr-) [reply services/tests/compiler.rs:206]: It's OK for test.

**Review by fzzr- — APPROVED** [review]

**Pull request merged**